server/lib/chat/chat-models.ts

```diff
@@ -88,19 +32,58 @@ function convertJoinedChannelFromDb(props: DbJoinedChannel): JoinedChannel {
   }
 }
 
-export async function getJoinedChannelForUser(
+/**
+ * Gets a user channel entry for each channel that a particular user is in, ordered by their channel
+ * join date.
+ */
+export async function getChannelsForUser(userId: SbUserId): Promise<UserChannelEntry[]> {
+  const { client, done } = await db()
+  try {
+    const result = await client.query<DbUserChannelEntry>(sql`
+      SELECT u.name AS user_name, c.*
+      FROM channel_users as c INNER JOIN users as u ON c.user_id = u.id
+      WHERE c.user_id = ${userId}
+      ORDER BY c.join_date;
+    `)
+    return result.rows.map(row => convertUserChannelEntryFromDb(row))
+  } finally {
+    done()
+  }
+}
+
+/**
+ * Gets a user info for each user in a particular channel. We don't order the users here since
+ * they're re-sorted alphabetically on the client anyway.
+ */
+export async function getUsersForChannel(channelName: string): Promise<SbUser[]> {
+  const { client, done } = await db()
+  try {
+    const result = await client.query<Dbify<SbUser>>(sql`
+      SELECT u.id, u.name
+      FROM channel_users as c INNER JOIN users as u ON c.user_id = u.id
+      WHERE c.channel_name = ${channelName};
+    `)
+    return result.rows
+  } finally {
+    done()
+  }
+}
+
+/**
+ * Gets a user channel entry for a particular user in a particular channel.
+ */
+export async function getChannelUserEntryForUser(
```

Comment:
Is this meant to be `getUserChannelEntryForUser`? As is it's a weird rearrangement from the actual type